interpreter.py

```diff
@@ -51,59 +73,111 @@ def exit_not_implemented(self, name):
         print "[ERROR]: Found unimplemented operation. Exiting."
         raise NotImplementedError(name)
 
-    def exec_operation(self, state, opcode, args=[]):
+    def puts(self, string, args=[]):
+        print string
+        if args:
+            print "Arguments are: "
+            for arg in args:
+                assert isinstance(arg, NumericType)
+                print arg.value
+            print "\n\n"
+
+    def exec_operation(self, function, state, instruction):
+        opcode = LLVMGetInstructionOpcode(instruction)
+        args = [LLVMGetOperand(instruction, i)\
+                for i in range(0,  LLVMGetNumOperands(instruction))]
         if opcode == LLVMRet:
-            return 0
+            return lookup_var(state, self.global_state, args[0])
         elif opcode == LLVMAdd:
             x, y = self._get_args(args, state)
-            return x + y
+            assert isinstance(x, NumericType) and isinstance(y, NumericType)
+            return Integer(x.value + y.value)
+        elif opcode == LLVMFAdd:
+            x, y = self._get_args(args, state)
+            assert isinstance(x, NumericType) and isinstance(y, NumericType)
+            return Float(x.value + y.value)
         elif opcode == LLVMMul:
             x, y = self._get_args(args, state)
-            return x * y
+            assert isinstance(x, NumericType) and isinstance(y, NumericType)
```

Comment:
why is this the more generic ``NumericType``, and not ``Integer``? It' can't be a ``Float``, right?